src/app/api/utils/auth.tsx

```diff
@@ -81,6 +81,33 @@ export const authOptions: AuthOptions = {
   },
   providers: [fxaProviderConfig],
   callbacks: {
+    // Do not pass un-parseable URLs to next-auth, @see MNTOR-3029
+    // Preserve default behavior otherwise. @see https://next-auth.js.org/configuration/callbacks#redirect-callback
+    redirect({ url, baseUrl }) {
+      // Allows relative callback URLs
+      if (url.startsWith("/")) {
+        if (URL.canParse(`${baseUrl}${url}`)) {
+          return `${baseUrl}${url}`;
+        } else {
+          logger.warn("nextauth_relative_redirect_url_parse_error", {
+            url: `${baseUrl}${url}`,
+          });
+          return baseUrl;
+        }
+      }
+      // Allows callback URLs on the same origin
+      else {
+        if (URL.canParse(url)) {
+          if (new URL(url).origin === baseUrl) {
+            return url;
+          }
+        } else {
+          logger.warn("nextauth_absolute_redirect_url_parse_error", { url });
+          return baseUrl;
+        }
+      }
+      return baseUrl;
```

Comment:
Reading [the docs](https://next-auth.js.org/configuration/callbacks#redirect-callback), it looks like the default behaviour is to only allow absolute URLs with the same base URL, and the relative URL bit is only to showcase the use of the callback?

```suggestion
      if (URL.canParse(url) && new URL(url).origin === baseUrl) {
        return url;
      }

      logger.warn("nextauth_absolute_redirect_url_parse_error", { url });
      return baseUrl;
```

Reply:
Hm I just took a look at what actually ships in the `next-auth` package and it looks like this:

```
 redirect({
    url,
    baseUrl
  }) {
    if (url.startsWith("/")) return `${baseUrl}${url}`;else if (new URL(url).origin === baseUrl) return url;
    return baseUrl;
  },
```